### PR TITLE
feat(watch): add ability to watch and auto-compile while using with serverless offline

### LINKED ISSUE #84: Won't auto compile with serverless offline
I could not get this plugin to auto compile on save with `serverless-offline` plugin.

Does the plugin support this function or I am missing something? 

I am running the plugin with the default configuration.

Thanks

### PR DESCRIPTION
**Pull request by adikari**
1. Add the ability to watch files and auto-compile while using with serverless offline.
2. Update document with usage. 

Closes #84

### CODE REVIEW

File: examples/complete/package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "typescript": "~3.9.7"
   },
   "dependencies": {
-    "isin-validator": "^1.1.1"
+    "isin-validator": "^1.1.1",
+    "serverless": "^2.17.0"
```
Comment (floydspace) [inline examples/complete/package.json:13]: It should be installed as dev dep

File: examples/minimal/package.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
     "serverless-offline": "^6.4.0"
   },
   "dependencies": {
-    "isin-validator": "^1.1.1"
+    "isin-validator": "^1.1.1",
+    "serverless": "^2.17.0"
```
Comment (floydspace) [inline examples/minimal/package.json:11]: same here

**Review by floydspace — APPROVED** [review]
hi @adikari thank you very much for your contribution, it's looking good.

**Comment by adikari** [comment]
Awesome. Let me know when this is merged.

**Comment by floydspace** [comment]
@adikari 👆

**Pull request merged**